serving/src/test/java/feast/serving/it/ServingServiceIT.java

```diff
@@ -0,0 +1,555 @@
+                java.time.Duration.ofMillis(2000)));
+    StatefulRedisConnection<byte[], byte[]> connection = redisClient.connect(new ByteArrayCodec());
+    syncCommands = connection.sync();
+
+    String projectName = "default";
+    // Apply Entity
+    String entityName = "driver_id";
+    ValueProto.Value entityValue = ValueProto.Value.newBuilder().setInt64Val(1).build();
+    String description = "My driver id";
+    ValueProto.ValueType.Enum entityType = ValueProto.ValueType.Enum.INT64;
+    EntityProto.EntitySpecV2 entitySpec =
+        EntityProto.EntitySpecV2.newBuilder()
+            .setName(entityName)
+            .setDescription(description)
+            .setValueType(entityType)
+            .build();
+    TestUtils.applyEntity(coreClient, projectName, entitySpec);
+
+    // Apply FeatureTable
+    String featureTableName = "rides";
+    List<String> entities =
+        new ArrayList<>() {
+          {
+            add(entityName);
+          }
+        };
+    HashMap<String, ValueProto.ValueType.Enum> features = new HashMap<>();
+
+    // Feature 1
+    String feature1 = "trip_cost";
+    ValueProto.Value feature1Value = ValueProto.Value.newBuilder().setDoubleVal(42.2).build();
+    ServingAPIProto.FeatureReferenceV2 feature1Reference =
+        ServingAPIProto.FeatureReferenceV2.newBuilder()
+            .setFeatureTable(featureTableName)
+            .setName(feature1)
+            .build();
+
+    // Feature 2
+    String feature2 = "trip_distance";
+    ValueProto.Value feature2Value = ValueProto.Value.newBuilder().setInt64Val(42).build();
+    ServingAPIProto.FeatureReferenceV2 feature2Reference =
+        ServingAPIProto.FeatureReferenceV2.newBuilder()
+            .setFeatureTable(featureTableName)
+            .setName(feature2)
+            .build();
+
+    // Feature 2
+    String feature3 = "trip_empty";
+    ValueProto.Value feature3Value = ValueProto.Value.newBuilder().build();
+    ServingAPIProto.FeatureReferenceV2 feature3Reference =
+        ServingAPIProto.FeatureReferenceV2.newBuilder()
+            .setFeatureTable(featureTableName)
+            .setName(feature3)
+            .build();
+
+    // Event Timestamp
+    String eventTimestampKey = timestampPrefix + ":" + featureTableName;
+    Timestamp eventTimestampValue = Timestamp.newBuilder().setSeconds(100).build();
+
+    features.put(feature1, ValueProto.ValueType.Enum.INT64);
+    features.put(feature2, ValueProto.ValueType.Enum.DOUBLE);
+    features.put(feature3, ValueProto.ValueType.Enum.DOUBLE);
+
+    TestUtils.applyFeatureTable(coreClient, projectName, featureTableName, entities, features);
+
+    // Serialize Redis Key with Entity i.e <default_driver_id_1>
+    redisKey =
+        RedisProto.RedisKeyV2.newBuilder()
+            .setProject(projectName)
+            .addEntityNames(entityName)
+            .addEntityValues(entityValue)
+            .build();
+
+    // Murmur hash Redis Value Field i.e murmur(<rides:trip_distance>)
```

Comment:
comment doesn't correspond to the code. Probably need to reorganize and split into functions